### DISCUSSION TITLE
Feature Request: Disable Highlighting/Selecting/Copying Row-Level Data

### DISCUSSION THREAD

**Discussion by bcook-konza** [apache/superset#29078] [Ideas]
Context:

As a user of Apache Superset, I work with sensitive or proprietary data displayed in query results. To enhance data security and prevent unauthorized copying or sharing of this sensitive information, I propose a feature that disables the ability to highlight, select, or copy row-level data displayed in the query results.

Feature Description:

Implement an optional setting in Apache Superset that, when enabled, prevents users from highlighting, selecting, and copying row-level data from query result tables. This setting should be configurable at both the dashboard and chart levels to provide flexibility in securing data.

Requirements/Ideas:

Configuration Option:

Add a configuration option in the Superset settings to enable or disable this feature.
Allow administrators to set this option at the dashboard level and individual chart level.

User Interface:

When the feature is enabled, modify the CSS of the query result tables to disable text selection.
Ensure the modification does not impact other functionalities like sorting, filtering, or pagination of the data.

Benefits:

Data Security: Enhances security for organizations dealing with sensitive data by reducing the risk of data leakage.
User Control: Provides administrators with better control over data visibility and handling within their Superset environment.

Implementation Suggestions:

Utilize CSS properties such as user-select: none; to prevent text selection.
Review and adjust JavaScript event handlers to ensure the feature works across all browsers and devices supported by Superset.

Conclusion:

This feature would be a valuable addition to Apache Superset, providing an extra layer of data protection for organizations handling sensitive information. Thank you for considering this request, and I am available to discuss further details or testing if needed.

**Comment by biakonza** [comment]
We are willing to implement this and provide a PR
